overlord/snapstate/check_snap.go

```diff
@@ -55,6 +54,15 @@ func checkAssumes(s *snap.Info) error {
 
 var openSnapFile = backend.OpenSnapFile
 
+// FIXME: drop and get the gadget name from the model assertion instead
+func isFirstBoot(st *state.State) bool {
+	chg := st.Change("1")
```

Comment:
We can easily have a different initial change without realizing we're breaking this.

I suggest dropping the use of isFirstBoot below, and adding a card for the RC to fix the race condition properly. During normal working, the remaining logic will catch and prevent the problem.